resources/build/minimum-versions.tests.inc.sh

```diff
@@ -0,0 +1,90 @@
+#
+# Keyman is copyright (C) SIL Global. MIT License.
+#
+# Created by mcdurdin on 2025-04-30
+#
+# Test that resources in our repository match the expected minimum versions,
+# where possible. Uses a variety of mechanisms to extract the version
+#
+
+. "${KEYMAN_ROOT}/resources/build/jq.inc.sh"
+
+#
+# Tests that the referenced filename has the expected version
+# found in the input variable. The actual version is extracted
+# with a helper function named _minver_test_$variable, which
+# parses the file or its metadata, as this may vary considerably
+#
+# Parameters
+#   1: variable    the name of the environment variable to test
+#   2: filename    the file to test the version against
+#
+_minver_test() {
+  local variable="$1"
+  local filename="$2"
+
+  local actual_version="$(_minver_test_${variable} "$filename")"
+  local -n expected_version="${variable}"
+
+  builder_echo "Testing \$$1, expecting '$expected_version'"
+  if [[ "$expected_version" != "$actual_version" ]]; then
+    builder_die "$filename: expected '$expected_version', actual version in repo is '$actual_version'"
+  fi
+}
+
+_minver_test_KEYMAN_VERSION_LANGUAGE_SUBTAG_REGISTRY() {
+  # language-subtag-registry format:
+  #   File-Date: 2025-03-10
+  head "$1" -n 1 | cut -d" " -f 2 -
+}
+
+_minver_test_KEYMAN_VERSION_UNICODE() {
+  # Blocks.txt format:
+  #   # Blocks-16.0.0.txt
+  head "$1" -n 1 | cut -d- -f 2 - | cut -d. -f 1,2,3 -
+}
+
+_minver_test_KEYMAN_VERSION_LANGTAGS() {
+  # entry with tag '_version'
+  "$JQ" -r -c '.[] | select(.tag == "_version") .date' "$1"
+}
+
+_minver_test_KEYMAN_VERSION_CLDR() {
+  # finds the folder with the expected name
+  ## find "$1" -name '[0-9][0-9]' -type d -printf '%f'
+  # note macos find does not support -printf, so went with this:
+  find "$1" -name '[0-9][0-9]' -type d -exec basename {} \;
+}
+
+_minver_test_KEYMAN_VERSION_ISO639_3() {
+  # git commit log for last commit for the file
+  git log -1 --format="%cs" -- "$1"
+}
+
+_minver_test_KEYMAN_VERSION_ICU() {
+  # source_filename = icu4c-73_1-src.tgz
+  grep "source_filename" "$1" | cut -d- -f 2 - | tr _ .
+}
```

Comment:
there's also a file uvernum.h with

```c
#define U_ICU_VERSION_MAJOR_NUM 73
```

Reply:
yes, I just picked one of the many :grin: